test/e2e/init/init_test.go

```diff
@@ -0,0 +1,229 @@
+				goldenFileAbsPath, err := filepath.Abs("golden/init_help.txt")
+				Expect(err).NotTo(HaveOccurred())
+				helper.CompareActualWithGolden(session, goldenFileAbsPath, true)
+			})
+
+			It("should exit with status code 0", func() {
+				session = helper.RunCommand(terrascanBinaryPath, outWriter, errWriter, initCommand, "-h")
+				Eventually(session).Should(gexec.Exit(0))
+			})
+		})
+
+		When("terrascan init command has typo. eg: inti", func() {
+			It("should print command suggestion", func() {
+				session = helper.RunCommand(terrascanBinaryPath, outWriter, errWriter, "inti")
+				goldenFileAbsPath, err := filepath.Abs("golden/init_typo_help.txt")
+				Expect(err).NotTo(HaveOccurred())
+				helper.CompareActualWithGolden(session, goldenFileAbsPath, false)
+			})
+
+			It("should exit with status code 1", func() {
+				session = helper.RunCommand(terrascanBinaryPath, outWriter, errWriter, "inti")
+				Eventually(session, 5).Should(gexec.Exit(1))
+			})
+		})
+
+		When("terrascan init is run with -c flag", func() {
+			Context("config file has valid policy config data", func() {
+				It("should download policies as per the policy config in the config file", func() {
+					Skip("skipping this test due to https://github.com/accurics/terrascan/issues/550, should be implemented when fixed")
+				})
+			})
+		})
+	})
+
+	Describe("terrascan init is run when TERRASCAN_CONFIG is set", func() {
+		When("the config file has invalid repo url", func() {
+			JustBeforeEach(func() {
+				os.Setenv(terrascanConfigEnvName, "config/invalid_repo.toml")
+			})
+			JustAfterEach(func() {
+				os.Setenv(terrascanConfigEnvName, "")
+			})
+			It("should error out and exit with status code 1", func() {
+				session = runInitCommand(terrascanBinaryPath, outWriter, errWriter, 1)
+				helper.ContainsErrorSubString(session, `failed to download policies. error: 'Get "https://repository/url/info/refs?service=git-upload-pack": dial tcp:`)
+			})
+		})
+		When("the config file has invalid branch name", func() {
+			JustBeforeEach(func() {
+				os.Setenv(terrascanConfigEnvName, "config/invalid_branch.toml")
+			})
+			JustAfterEach(func() {
+				os.Setenv(terrascanConfigEnvName, "")
+			})
+			It("should error out and exit with status code 1", func() {
+				session = runInitCommand(terrascanBinaryPath, outWriter, errWriter, 1)
+				helper.ContainsErrorSubString(session, `failed to initialize terrascan. error : failed to checkout git branch 'invalid-branch'. error: 'reference not found'`)
+			})
+		})
+		When("the config file has invalid rego subdir", func() {
+			JustBeforeEach(func() {
+				os.Setenv(terrascanConfigEnvName, "config/invalid_rego_subdir.toml")
+			})
+			JustAfterEach(func() {
+				os.Setenv(terrascanConfigEnvName, "")
+			})
+			It("should error out and exit with status code 1", func() {
+				session = runInitCommand(terrascanBinaryPath, outWriter, errWriter, 1)
+				helper.ContainsErrorSubString(session, "invalid/path: no such file or directory")
```

Comment:
we should make changes to return a more specific error message for erranous rego subdir.
Not to be fixed in this PR though.